### PR TITLE
add concent_enabled flag to task definition

### LINKED ISSUE #3122: [Concent] Add `concent_enabled` flag to the task header
`TaskFixedHeader` dictionary needs to include the `concent_enabled` flag

### PR DESCRIPTION
**Pull request by etam**
Resolves #3122 #3075

### CODE REVIEW

File: golem/client.py [hunk 1]
```diff
@@ -565,12 +565,18 @@ def enqueue_new_task(self, task_dict) -> Tuple[Deferred, str]:
...
+        task: TaskBase
         if isinstance(task_dict, dict):
             logger.warning('enqueue_new_task called with deprecated dict type')
             task = task_manager.create_task(task_dict)
         else:
             task = task_dict
 
+        if task.header.fixed_header.concent_enabled and \
+                not self.concent_service.enabled:
+            raise Exception("Cannot create task with concent enabled when "
```
Comment (jiivan) [inline golem/client.py]: Is it possible to use a more specific Exception here?

**Review by jiivan — COMMENTED** [review]
LGTM, except I would like to see more specific exception when concent is requested but not available.

File: golem/client.py (hunk 1 above)
Comment (etam) [reply golem/client.py]: Of course it is. Should I use `RuntimeError` or a custom exception class?

Comment (jiivan) [reply golem/client.py]: I vote for a custom exception.

**Review by jiivan — APPROVED** [review]

**Pull request merged**